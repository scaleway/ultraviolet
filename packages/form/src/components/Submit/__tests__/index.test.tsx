import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm } from '@utils/test'
import { describe, expect, test } from 'vitest'
import { Submit, TextInputField } from '../..'

const alpha = /^[a-zA-Z]*$/

describe('Submit', () => {
  test('renders correctly ', () => {
    const { asFragment } = renderWithForm(<Submit>Test</Submit>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('form is invalid', () => {
    const { asFragment } = renderWithForm(
      <>
        <TextInputField label="test" name="toto" regex={[alpha]} />
        <Submit>Test</Submit>
      </>,
      { defaultValues: { toto: '4' } },
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('form is submitting', async () => {
    const { asFragment } = renderWithForm(
      <Submit>Test</Submit>,
      {},
      {
        onSubmit: () =>
          new Promise(resolve => {
            setTimeout(() => resolve(undefined), 500)
          }),
      },
    )
    await userEvent.click(
      // eslint-disable-next-line testing-library/no-node-access
      screen.getByText('Test').closest('button') as HTMLButtonElement,
    )
    expect(
      // eslint-disable-next-line testing-library/no-node-access
      screen.getByText('Test').closest('button') as HTMLButtonElement,
    ).toBeDisabled()

    expect(asFragment()).toMatchSnapshot()
  })
})
