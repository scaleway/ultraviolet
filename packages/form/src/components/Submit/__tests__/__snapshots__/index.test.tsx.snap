// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Submit > form is invalid 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: contents;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.emotion-5 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-7>.emotion-10 {
  color: #3f4250;
}

.emotion-7>.emotion-10::-webkit-input-placeholder {
  color: #727683;
}

.emotion-7>.emotion-10::-moz-placeholder {
  color: #727683;
}

.emotion-7>.emotion-10:-ms-input-placeholder {
  color: #727683;
}

.emotion-7>.emotion-10::placeholder {
  color: #727683;
}

.emotion-7[data-success='true'] {
  border-color: #22674e;
}

.emotion-7[data-error='true'] {
  border-color: #b3144d;
}

.emotion-7[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-7[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-7[data-disabled='true']>.emotion-10 {
  color: #b5b7bd;
}

.emotion-7[data-disabled='true']>.emotion-10::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-7[data-disabled='true']>.emotion-10::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-7[data-disabled='true']>.emotion-10:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-7[data-disabled='true']>.emotion-10::placeholder {
  color: #b5b7bd;
}

.emotion-7:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-9 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-9[data-size='large'] {
  font-size: 1rem;
}

.emotion-9[data-size='small'] {
  padding-left: 0.5rem;
}

<div
    data-testid="testing"
  >
    <form
      class="emotion-0 emotion-1"
      novalidate=""
    >
      <div
        class="emotion-2 emotion-3"
      >
        <label
          class="emotion-4 emotion-5 emotion-6"
          for="«r1»"
        >
          test
        </label>
        <div>
          <div
            class="emotion-7 emotion-8"
            data-disabled="false"
            data-error="false"
            data-readonly="false"
            data-success="false"
          >
            <input
              aria-invalid="false"
              class="emotion-9 emotion-10"
              data-size="large"
              id="«r1»"
              name="toto"
              type="text"
              value="4"
            />
          </div>
        </div>
      </div>
      <button
        class=" emotion-11 emotion-12 emotion-13 e1wcoej e1wcoeg emotion-14 e1wcoel"
        disabled=""
        type="submit"
      >
        Test
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Submit > form is submitting 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: contents;
}

.emotion-6 {
  -webkit-animation: spin 0.75s linear infinite;
  animation: spin 0.75s linear infinite;
}

.emotion-8 {
  stroke: #d9dadd;
}

.emotion-10 {
  -webkit-transition: stroke-dashoffset 0.5s ease 0s;
  transition: stroke-dashoffset 0.5s ease 0s;
}

<div
    data-testid="testing"
  >
    <form
      class="emotion-0 emotion-1"
      novalidate=""
    >
      <button
        class=" emotion-2 emotion-3 emotion-4 e1wcoej e1wcoeg emotion-5 e1wcoel"
        disabled=""
        type="submit"
      >
        <svg
          aria-label="Loading"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="20"
          aria-valuetext="20%"
          class="emotion-6 emotion-7"
          role="progressbar"
          style="height: 1rem; width: 1rem;"
          viewBox="0 0 100 100"
        >
          <circle
            class="emotion-8 emotion-9"
            cx="50"
            cy="50"
            fill="none"
            r="42"
            stroke-width="16"
          />
          <circle
            class="emotion-10"
            cx="50"
            cy="50"
            fill="none"
            r="42"
            stroke="#ffffff"
            stroke-dasharray="263.89378290154264"
            stroke-dashoffset="211.11502632123413"
            stroke-linecap="round"
            stroke-width="16"
          />
        </svg>
        Test
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Submit > renders correctly  1`] = `
<DocumentFragment>
  .emotion-0 {
  display: contents;
}

<div
    data-testid="testing"
  >
    <form
      class="emotion-0 emotion-1"
      novalidate=""
    >
      <button
        class=" emotion-2 emotion-3 emotion-4 e1wcoej e1wcoeg emotion-5 e1wcoel"
        disabled=""
        type="submit"
      >
        Test
      </button>
    </form>
  </div>
</DocumentFragment>
`;
